### PR TITLE
FOLIO-3448 Javadoc for FolioExecutionContext is added

### PR DESCRIPTION
**Pull request by MikhailFokanov**
## Purpose
Javadoc and readme section are added to inform developers about appropriate usage of FolioExecutionContext.

### CODE REVIEW

**Review by TarasSpashchenko — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -27,6 +27,19 @@ Version 2.0. See the file "[LICENSE](LICENSE)" for more information.
...
+[FolioExecutionContext](src/main/java/org/folio/spring/FolioExecutionContext.java) is used to store
+essential request headers <i>(in thread local)</i>. Folio Spring Base populates this data
+using [FolioExecutionScopeFilter](src/main/java/org/folio/spring/scope/filter/FolioExecutionScopeFilter.java).
+It is used by [EnrichUrlAndHeadersClient](src/main/java/org/folio/spring/client/EnrichUrlAndHeadersClient.java), to provide right tenant id and other headers for outgoing REST requests.
+It is also used in [DataSourceSchemaAdvisorBeanPostProcessor](src/main/java/org/folio/spring/config/DataSourceSchemaAdvisorBeanPostProcessor.java) for selection of the appropriate schema for sql queries.
+
+FolioExecutionContext is immutable. In order to start new execution context `FolioExecutionScopeExecutionContextManager.beginFolioExecutionContext(new DefaultFolioExecutionContext(...))` should be used.
+
+***CAUTION: FolioExecutionContext should not be used in asynchronous code executions (as it is stored in thread local), unless
+the appropriate data is manually set by `beginFolioExecutionContext(...)`***
```
Comment (julianladisch) [inline README.md:41]: Should `endFolioExecutionContext()` been called at the end of the async code?

File: README.md [hunk 2]
```diff
@@ -27,6 +27,19 @@ Version 2.0. See the file "[LICENSE](LICENSE)" for more information.
...
+
```
Comment (julianladisch) [inline README.md:42]: Can you provide some code snippet that shows an asynchronous code execution that properly sets FolioExecutionContext?

**Review by dmtkachenko — APPROVED** [review]

**Comment by MikhailFokanov** [comment]
@julianladisch, could you please see my fixes for your comments?

**Review by julianladisch — APPROVED** [review]
Looks good, thanks!

**Pull request merged**